src/core.jl

```diff
@@ -8,3 +8,9 @@ immutable SegmentedImage{T<:AbstractArray,U<:Colorant}
     segment_means::Dict{Int,U}
     segment_pixel_count::Dict{Int,Int}
 end
+
+struct ImageEdge
```

Comment:
If a user is constructing a RAG manually (which is part of your intended API), is this sufficiently helpful? I like the following syntax:
```julia
"""
    edge = ImageEdge(index1, index2, weight)

Construct an edge in a Region Adjacency Graph. `index1` and `index2` are the integers corresponding to individual pixels/voxels (in the sense of linear indexing via `sub2ind`), and `weight` ...
"""
```